src/test/java/com/example/javaOjt/controllers/AuthorControllerTest.java

```diff
@@ -29,7 +34,143 @@ class AuthorControllerTest {
+        .andReturn();
+
+    // Verify that the service method was called with the correct parameters
+    Mockito.verify(authorService, Mockito.times(1)).getAuthorsList(AuthorSortBy.NAME, null, null);
+  }
+
+  @Test
+  void getAuthorsList_order() throws Exception {
+    // Mocking the service response
+    GetAuthorsResponse mockResponse = new GetAuthorsResponse();
+    Mockito.when(authorService.getAuthorsList(null, Order.DSC, null))
+        .thenReturn(mockResponse);
+
+    // Perform the GET request
+    mockMvc.perform(
+            MockMvcRequestBuilders.get(AUTHOR_BASE_URL).param("order", "dsc"))
+        .andExpect(MockMvcResultMatchers.status().isOk())
+        .andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON))
+        .andReturn();
+
+    // Verify that the service method was called with the correct parameters
+    Mockito.verify(authorService, Mockito.times(1)).getAuthorsList(null, Order.DSC, null);
+  }
+
+  @Test
+  void getAuthorsList_illegalAttribute() throws Exception {
+    // Perform the GET request
+    mockMvc.perform(
+            MockMvcRequestBuilders.get(AUTHOR_BASE_URL)
+                .param("sort_by", "v50"))
+        .andExpect(MockMvcResultMatchers.status().isBadRequest())
+        .andReturn();
+
+    // Verify that the service method was called with the correct parameters
+    Mockito.verify(authorService, Mockito.never())
+        .getAuthorsList(Mockito.any(AuthorSortBy.class), Mockito.any(Order.class),
+            Mockito.anyList());
+  }
+
+  @Test
+  void getAuthorsList_illegalOrder() throws Exception {
+    // Perform the GET request
+    mockMvc.perform(
+            MockMvcRequestBuilders.get(AUTHOR_BASE_URL)
+                .param("order", "kfc"))
+        .andExpect(MockMvcResultMatchers.status().isBadRequest())
+        .andReturn();
+
+    // Verify that the service method was called with the correct parameters
+    Mockito.verify(authorService, Mockito.never())
+        .getAuthorsList(Mockito.any(AuthorSortBy.class), Mockito.any(Order.class),
+            Mockito.anyList());
+  }
+
+  @Test
+  void getAuthorsList_ids() throws Exception {
+    List<Integer> targetIds = List.of(1, 2);
+    // Mocking the service response
+    GetAuthorsResponse mockResponse = new GetAuthorsResponse();
+    Mockito.when(authorService.getAuthorsList(null, null, targetIds))
+        .thenReturn(mockResponse);
+
+    // Perform the GET request
+    mockMvc.perform(
+            MockMvcRequestBuilders.get(AUTHOR_BASE_URL).param("ids", "1", "2"))
+        .andExpect(MockMvcResultMatchers.status().isOk())
+        .andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON))
+        .andReturn();
+
+    // Verify that the service method was called with the correct parameters
+    Mockito.verify(authorService, Mockito.times(1)).getAuthorsList(null, null, targetIds);
+  }
+
+  @Test
+  void getAuthorsList_invalidId() throws Exception {
```

Comment:
細かいですが、このテストは「存在しない ids」を指定するテストだと思いますので、

```suggestion
  void getAuthorsList_notExistedId() throws Exception {
```